internal/app/connectconformance/server_runner.go

```diff
@@ -26,8 +26,145 @@ import (
 	"connectrpc.com/conformance/internal"
 	conformancev1alpha1 "connectrpc.com/conformance/internal/gen/proto/go/connectrpc/conformance/v1alpha1"
 	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 )
 
+// populateExpectedRequests populates the requests we expected to get back from the server
+// by examining the requests we sent.
+func populateExpectedRequests(testCase *conformancev1alpha1.TestCase) error {
```

Comment:
I think there's enough logic here that we want a test. Otherwise, there's the risk of a bug that means we may fail to catch certain issues with the response.

Reply:
Also, this should probably be called `populateExpectedResponse`, since the name of the field it populates is `ExpectedResponse`, not `ExpectedRequests`.

Reply:
Renamed and added tests. I apologize in advance for the size of it. Not sure if there's a better way. There are a lot of cases to cover and a lot of structs to create to do so 😬